### PR TITLE
Setting to ignore pubspec.yaml in bazel projects

### PR DESCRIPTION
**Pull request by DaveShuckerow**
This makes the Flutter plugin obey the existing registry setting "dart.projects.without.pubspec" from the Dart plugin.

### CODE REVIEW

File: src/io/flutter/settings/FlutterSettings.java [hunk 1]
```diff
@@ -217,6 +227,25 @@ public void setOpenInspectorOnAppLaunch(boolean value) {
...
+    fireEvent();
+  }
+
+  /**
+   * Determines whether bazel should be used for a project based on the user's preferences and the structure of the project.
+   */
+  public boolean shouldUseBazel(Project project) {
+    final boolean isBazelProject = FlutterModuleUtils.isFlutterBazelProject(project);
+    final boolean isNotPubProject = PubRoots.forProject(project).isEmpty();
+    return isBazelProject && (useBazelByDefault() || isNotPubProject);
```
Comment (jwren) [inline src/io/flutter/settings/FlutterSettings.java]: Why not just `return Registry.is(ignorePubspecRegistryKey, false);`  Is there some usecase where this complexity will help?

File: src/io/flutter/settings/FlutterSettings.java [hunk 2]
```diff
@@ -26,6 +30,9 @@
   private static final String syncAndroidLibrariesKey = "io.flutter.syncAndroidLibraries";
   private static final String disableTrackWidgetCreationKey = "io.flutter.disableTrackWidgetCreation";
   private static final String useFlutterLogView = "io.flutter.useLogView";
+  // The Dart plugin uses this registry key to keep bazel users from getting their settings overridden on projects that include a
+  // pubspec.yaml.
+  private static final String ignorePubspecRegistryKey = "dart.projects.without.pubspec";
```
Comment (jwren) [inline src/io/flutter/settings/FlutterSettings.java]: Rename to something like `DART_PROJECTS_WITHOUT_PUBSPEC_KEY`, the current name makes it sound like you are going to ignore the value stored in `dart.projects.without.pubspec`.

Comment (DaveShuckerow) [reply src/io/flutter/settings/FlutterSettings.java]: Good idea, done.

File: src/io/flutter/settings/FlutterSettings.java (hunk 1 above)
Comment (DaveShuckerow) [reply src/io/flutter/settings/FlutterSettings.java]: Yes, this is useful:

 - If we ask for the value of the setting in a pub-only project, then we should not attempt to use bazel.
 - Conversely, if this is not set in a bazel-only project, then we should still always use bazel.

File: src/io/flutter/sdk/FlutterSettingsConfigurable.form [hunk 3]
```diff
@@ -88,6 +88,15 @@
               <toolTipText value="Enables verbose logging (this can be useful for diagnostic purposes)."/>
             </properties>
           </component>
+          <component id="984f6" class="javax.swing.JCheckBox" binding="myUseBazelByDefaultCheckBox">
+            <constraints>
+              <grid row="3" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="3" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+            </constraints>
+            <properties>
+              <text value="Use Bazel instead of Pub by default"/>
```
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterSettingsConfigurable.form]: Do we need a UI setting for this? Should this be controlled on a per app basis, or just via the registry setting?

**Comment by devoncarew** [comment]
Dave, why don't you land this first, and then I can land https://github.com/flutter/flutter-intellij/pull/3436 (we'll have merge conflicts with these two PRs).

File: src/io/flutter/sdk/FlutterSettingsConfigurable.form (hunk 3 above)
Comment (DaveShuckerow) [reply src/io/flutter/sdk/FlutterSettingsConfigurable.form]: I think ideally, we will be able to determine this and other sensible defaults for Bazel at startup time and apply them in bulk.

Until adding that, this is useful.  I filed #3437 to do this, and I'll submit this change for now.

**Pull request merged**